### ISSUE TITLE
OperationID from embedded spec doesn't match the original spec

### ISSUE THREAD

**Issue by skoo25** [deepmap/oapi-codegen#1274]
Version: v1.15

I'm not sure if this is by design, but given an OAPI spec
```yaml
openapi: "3.0.0"
info:
  title: "my spec"
  version: 1.0.0
paths:
  /pet:
    get:
      operationId: getPet
      responses:
        200:
          content:
            application/json:
              schema:
                type: string
```
OperationId from the embedded spec is set as `GetPet` rather than `getPet`. I used the generated `GetSwagger()` function to convert embedded spec to `openapi3.T` object.

**Comment by nickajacks1** [comment]
this capitalization is done explicitly, see ToCamelCase in the codegen package.
This was probably done so that the generated function always gets exported. Users would probably find it surprising if the capitalization of the operation ID in their spec controlled whether the resulting function is exported or not.

**Comment by skoo25** [comment]
> this capitalization is done explicitly, see ToCamelCase in the codegen package. This was probably done so that the generated function always gets exported. Users would probably find it surprising if the capitalization of the operation ID in their spec controlled whether the resulting function is exported or not.

Indeed. I'm trying to write a middleware utilizing Operation ID specified within the embedded spec. However, the capitalization proves to be a challenge, as the ID is case-sensitive according to the OAPI specification.

**Comment by nickajacks1** [comment]
Right now there's a generation option that controls capitalization of initialims. Perhaps there could be single option that controls various modes of capitalization, including force-camel-case, force-camel-case with Go initialisms, and a new _preserve casing_ option.  Then in v2 the initialisms specific parameter could be removed. @jamietanna any thoughts on adding a new parameter for strict case preservation?

From the Openapi 3.1.0 spec:
> Tools and libraries MAY use the operationId to uniquely identify an operation, therefore, it is RECOMMENDED to follow common programming naming conventions.

From this, it sounds like OpenAPI recommends we write operation IDs with Go name conventions in mind, so "MyOpID" instead of "myOpId".  Of course, that complicates things a tad when you access a single API is shared between languages..

**Comment by jamietanna** [comment]
I think a good option here - with the current state of the codebase - would be to add a "name normalizer" which is a No-Op

**Comment by jamietanna** [comment]
This should now be fixed under https://github.com/oapi-codegen/oapi-codegen/pull/1945 which introduces a new configuration option to opt-in to the behaviour.

This is also something that I would see as being a good option to be a default behaviour as of `oapi-codegen` v3 (if we ever get there!)
